### DISCUSSION TITLE
Project building

### DISCUSSION THREAD

**Discussion by mbaimuratov** [robertodoering/harpy#773] [Q&A] [answered]
Hey, guys!

I am trying to build this project in Android Studio, but I am getting this error when trying to do `flutter pub get` in the project:

> Because every version of flutter_test from sdk depends on material_color_utilities 0.2.0 and harpy depends on material_color_utilities ^0.1.4, flutter_test from sdk is forbidden.
So, because harpy depends on flutter_test from sdk, version solving failed.
pub get failed
command: "/Users/mac/flutter/flutter/bin/cache/dart-sdk/bin/dart __deprecated_pub --verbosity=warning get --no-precompile"
pub env: {
  "FLUTTER_ROOT": "/Users/mac/flutter/flutter",
  "PUB_ENVIRONMENT": "flutter_cli:get",
  "PUB_CACHE": "/Users/mac/.pub-cache",
}
exit code: 1
last line of pub output: "So, because harpy depends on flutter_test from sdk, version solving failed."

And when I change `material_color_utilities 0.1.4` to `material_color_utilities 0.2.0` and try to do `flutter packages pub run build_runner build`, I get this:

> os ~/StudioProjects/harpy [master *=]$ flutter packages pub run build_runner build
Failed to build build_runner:build_runner:
../../.pub-cache/hosted/pub.dartlang.org/file-6.1.2/lib/src/interface/file.dart:15:16: Error: The method 'File.create' has fewer named arguments than those of overridden method 'File.create'.
  Future<File> create({bool recursive = false});
               ^
../../flutter/flutter/bin/cache/dart-sdk/lib/io/file.dart:232:16: Context: This is the overridden method ('create').
  Future<File> create({bool recursive = false, bool exclusive = false});
               ^
../../.pub-cache/hosted/pub.dartlang.org/file-6.1.2/lib/src/interface/file.dart:15:16: Error: The method 'File.create' doesn't have the named parameter 'exclusive' of overridden method 'File.create'.
  Future<File> create({bool recursive = false});
               ^
../../flutter/flutter/bin/cache/dart-sdk/lib/io/file.dart:232:16: Context: This is the overridden method ('create').
  Future<File> create({bool recursive = false, bool exclusive = false});
               ^
../../.pub-cache/hosted/pub.dartlang.org/file-6.1.2/lib/src/forwarding/forwarding_file.dart:22:8: Error: The method 'ForwardingFile.createSync' has fewer named arguments than those of overridden method 'File.createSync'.
  void createSync({bool recursive = false}) =>
       ^
../../flutter/flutter/bin/cache/dart-sdk/lib/io/file.dart:248:8: Context: This is the overridden method ('createSync').
  void createSync({bool recursive = false, bool exclusive = false});
       ^
../../.pub-cache/hosted/pub.dartlang.org/file-6.1.2/lib/src/forwarding/forwarding_file.dart:22:8: Error: The method 'ForwardingFile.createSync' doesn't have the named parameter 'exclusive' of overridden method 'File.createSync'.
  void createSync({bool recursive = false}) =>
       ^
../../flutter/flutter/bin/cache/dart-sdk/lib/io/file.dart:248:8: Context: This is the overridden method ('createSync').
  void createSync({bool recursive = false, bool exclusive = false});
       ^
../../.pub-cache/hosted/pub.dartlang.org/file-6.1.2/lib/src/backends/memory/memory_file.dart:57:8: Error: The method 'MemoryFile.createSync' has fewer named arguments than those of overridden method 'File.createSync'.
  void createSync({bool recursive = false}) {
       ^
../../flutter/flutter/bin/cache/dart-sdk/lib/io/file.dart:248:8: Context: This is the overridden method ('createSync').
  void createSync({bool recursive = false, bool exclusive = false});
       ^
../../.pub-cache/hosted/pub.dartlang.org/file-6.1.2/lib/src/backends/memory/memory_file.dart:57:8: Error: The method 'MemoryFile.createSync' doesn't have the named parameter 'exclusive' of overridden method 'File.createSync'.
  void createSync({bool recursive = false}) {
       ^
../../flutter/flutter/bin/cache/dart-sdk/lib/io/file.dart:248:8: Context: This is the overridden method ('createSync').
  void createSync({bool recursive = false, bool exclusive = false});
       ^
pub finished with exit code 1
os ~/StudioProjects/harpy [master *=]$ 

Could someone help me with this, so that I could start contributing to this awesome project? :)

**Comment by robertodoering** [comment, marked as answer]
Make sure you are on the latest flutter stable when working on harpy.
`flutter channel stable && flutter upgrade`

**Reply by mbaimuratov** [reply]
Working now! Now, I'm waiting for developer application approval from Twitter to start working :(
Don't you have some test api keys to authenticate into the app?

**Reply by robertodoering** [reply]
I cannot provide any test credentials, unfortunately

**Reply by robertodoering** [reply]
@mbaimuratov
I can give you a test key after all - but only for development. If you are still interested in contributing to harpy, feel free to send me an email and I will give you the key.

**Reply by mbaimuratov** [reply]
meirbek.baimuratov@gmail.com
Will be glad to start developing!
